### PR TITLE
SASS 3251 - Update All Pension Data to include pension income 

### PR DESCRIPTION
**Pull request by abiodxn**
# Checklist

Jira subtask: SASS-3248

## Developer: Before Asking Others To Review Your Code

Have you ...

- [x]  Had a QA verify that all acceptance criteria have been met, with comment(s) in the ticket? 
- [x]  Hosted a group review of your implementation?
- [x]  Rebased against `main`? 
- [x]  Ensured that your commit(s) contain the ticket key?
- [x]  Ensured that your branch is named after the ticket key?
- [x]  Successfully run the unit tests?
- [x]  Successfully run the integration tests?
- [x]  Successfully run the journey tests? If not, please add a comment explaining why it wasn't necessary.
- [x]  Ensured that the PR job has passed.

## Reviewer: Before Reviewing

Have you ...

- [x]  Pulled the branch down?
- [x]  Assigned yourself to this PR?
- [x]  Moved the Jira *subtask* to “in review”?
- [x]  Ensured that it is rebased against `main`?

## Developer: After Approval

Have you ...

- [ ]  Rebased against `main`?
- [ ]  Moved the Jira *subtask* to “in pipeline”?

## Developer: After Merging

Have you ...

- [ ]  Deleted the branch?
- [ ]  Scheduled a demo? If not, please add a comment.

### CODE REVIEW

**Comment by erasmos-hmrc** [comment]
@abiodxn , could you amend the commit message to contain the parent key (SASS-3248)?

**Review by salman-a-dev — APPROVED** [review]

File: test/builders/PaymentsIntoOverseasPensionsViewModelBuilder.scala [hunk 1]
```diff
@@ -16,25 +16,24 @@
...
-    taxReliefQuestion = None,
+    paymentsIntoOverseasPensionsQuestions = Some(true),
+    paymentsIntoOverseasPensionsAmount = Some(1999.99),
+    employerPaymentsQuestion = Some(true),
+    taxPaidOnEmployerPaymentsQuestion = Some(false),
+    customerReferenceNumberQuestion = Some("PENSIONINCOME245"),
+    employerPaymentsAmount = Some(1999.99),
+    taxReliefQuestion = Some(TaxReliefQuestion.TransitionalCorrespondingRelief),
     qualifyingOverseasPensionSchemeReferenceNumber = None,
     doubleTaxationCountryCode = None,
```
Comment (srinivaspach) [inline test/builders/PaymentsIntoOverseasPensionsViewModelBuilder.scala:32]: Any reason why these are left as None?

**Review by srinivaspach — CHANGES_REQUESTED** [review]
Looks good to me

File: app/models/pension/AllPensionsData.scala [hunk 2]
```diff
@@ -106,8 +108,32 @@ object AllPensionsData {
         pensionSchemeTaxReference = prior.pensionCharges.flatMap(_.pensionSchemeUnauthorisedPayments.map(_.pensionSchemeTaxReference))
       ),
 
-      //TODO Implement PaymentsIntoOverseas Endpoint
-      paymentsIntoOverseasPensions = PaymentsIntoOverseasPensionsViewModel())
+    paymentsIntoOverseasPensions = PaymentsIntoOverseasPensionsViewModel(
```
Comment (erasmos-hmrc) [inline app/models/pension/AllPensionsData.scala:111]: As mentioned during group review, there's much duplication.

The alternative I had in mind was something like this:

```
private def paymentsIntoOverseasPensionsViewModel(prior: AllPensionsData): PaymentsIntoOverseasPensionsViewModel = (prior.pensionIncome, prior.pensionReliefs) match {
    case (Some(pensionsIncome), Some(pensionsReliefs)) => paymentsIntoOverseasPensionsViewModel(pensionsIncome, pensionsReliefs.pensionReliefs)
    case (Some(pensionsIncome), None) => paymentsIntoOverseasPensionsViewModel(pensionsIncome)
    case (None, Some(pensionsReliefs)) => paymentsIntoOverseasPensionsViewModel(pensionsReliefs.pensionReliefs)
    case (None, None) => paymentsIntoOverseasPensionsViewModel()
  }
```

However, given the limited time, best to leave it like this.

**Review by srinivaspach — APPROVED** [review]
Looks good to me

**Review by srinivaspach — APPROVED** [review]

**Pull request merged**